core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -237,6 +237,8 @@ class SparkHadoopUtil extends Logging {
     }
   }
 
+  def pathExists( p: Path, config: Configuration): Boolean = p.getFileSystem(config).exists(p)
```

Comment:
There are some style issues in these changes like extra spaces before `p`. Also I don't think it's useful to make this method, since you use it once, and it isn't replacing the other 30-40 instances of this. It doesn't add much.